apps/courses/src/routes/users/dashboard/Mentor.tsx

```diff
@@ -352,27 +352,67 @@ export const MentorTabs = ({ courses, mentor }) => {
     );
   };
 
-  const MyActivity = () => {
-    const user: firebase.User = useUser();
-    const db = useFirestore();
-    const dbReviewsRef = db
+  const loadReviewData = async (db, uId, startAt, pageSize) => {
+    let dbRef = db
       .collection('users')
-      .doc(user.uid)
+      .doc(uId)
       .collection('reviews')
       // .where('status', '!=', 'pending')
       .orderBy('started_at', 'desc')
-      .limit(10);
-    const dbReviews: MentorReview[] = useFirestoreCollectionData(dbReviewsRef);
+      .limit(pageSize + 1);
+    if (startAt) {
+      dbRef = dbRef.startAt(startAt);
+    }
+    const reviewsSnapshots = await dbRef.get();
+    const reviews = reviewsSnapshots.docs.map((doc) => doc.data());
+
+    if (reviews.length > pageSize) {
+      return {
+        reviews: reviews.slice(0, pageSize),
+        lastReview: reviews[pageSize],
+      };
+    } else {
+      return {
+        reviews,
+        lastReview: null,
+      };
+    }
+  };
+
+  const MyActivity = () => {
+    const PAGE_SIZE = 5;
+    const user: firebase.User = useUser();
+    const db = useFirestore();
+
+    const [pageLoading, setPageLoading] = useState<boolean>(false);
+    const [dbReviews, setDbReviews] = useState<MentorReview[]>([]);
+    const [lastReview, setLastReiew] = useState<MentorReview>(null);
+
+    const loadNextPage = async () => {
+      setPageLoading(true);
+      const { lastReview: newLastReview, reviews: newReviews } = await loadReviewData(
+        db,
+        user.uid,
+        lastReview,
```

Comment:
In `loadReviewData`, the query orders by `started_at`. Should here be `lastReview.started_at` instead of `lastReview`?